tests/cpp/test_gpu2.cpp

```diff
@@ -2648,11 +2581,12 @@ TEST_P(WelfordReduction, Test) {
     outputs_of_red.push_back(M2_cast);
   }
 
-  auto rparams = getReductionHeuristics(&fusion, {aten_input});
-  scheduleReduction(&fusion, rparams.get());
+  auto heuristic_params = SchedulerEntry::scheduleWith(
```

Comment:
It didn't work, there's a comment below which might be why. I wasn't sure why it didn't work but I didn't want to dive into it at this time.

Reply:
Ah, it may be actually necessary to pass `CompileParams` to `FusionExecutor::compileFusion` in `scheduleAndRun`. It isn't used very often, so most of the tests don't need it, but that matters for this case.

Reply:
```  // Needs to pass compile para to use the correct index type, otherwise the
  // lowering pass will use int64 as the index tpye, since this test saves
  // `tv_N` as index type, it may cause vectorization size validation error. For
  // example, the heuristics set index type to int32 and the max vectorization
  // factor is 4, if compile para is not passed to compileFusion, the lowering
  // pass uses int64 as index type, so the max vectorization factor is 16 bytes
  // sizeof(int64) = 2, which is wrong since the actual index type is int32
  // and the max vectorization factor is 4.```